### PR TITLE
Add new expand_shell argument for run_command, to disable expanding shellisms

### LINKED ISSUE #45418: module.run_command() doesn't allow to escape $ when use_unsafe_shell=False
##### SUMMARY
`module.run_command()` doesn't allow to escape `$` in command when `use_unsafe_shell=False`.

Let's say I need to execute a command with a generated password as an argument:

```python
cmd = ['openssl', 'genrsa', '-aes128', '-passout', 'pass:(lFjq$_!', '-out', 'privkey.pem', '2048']
module.run_command(cmd)
```

Environment variables references (including *accidental* ones) will be expanded. For examble: `pass:(lFjq$_!` will be expanded to something like `pass:(lFjq/path/to/ansible-playbook!`, which is obviously not what is expected.

This is because `module.run_command()` calls `os.path.expandvars()` on each command arg. There's no way to escape `$` sign.

Setting `use_unsafe_shell=True` works fine, since it doesn't expand variables. But this doesn't seem consistent.

IMO, `module.run_command()` shouldn't call `os.path.expandvars()`. Doing variable expansion should be up to the caller.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
module_utils

##### ANSIBLE VERSION
```
ansible 2.5.4
  config file = None
  configured module search path = [u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/share/ansible/lib/ansible
  executable location = /usr/share/ansible/bin/ansible
  python version = 2.7.14 (default, Sep 23 2017, 22:06:14) [GCC 7.2.0]
```

##### CONFIGURATION
N/A

##### OS / ENVIRONMENT
N/A

##### STEPS TO REPRODUCE
```python
cmd = ['openssl', 'genrsa', '-aes128', '-passout', 'pass:(lFjq$_!', '-out', 'privkey.pem', '2048']
module.run_command(cmd)
```

##### EXPECTED RESULTS
Environment variables should not be expanded.

##### ACTUAL RESULTS
Environment variables are expanded.

### PR DESCRIPTION
**Pull request by sivel**
##### SUMMARY
Add new expand_shell argument for run_command, to disable expanding shellisms. Fixes #45418

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
lib/ansible/module_utils/basic.py

##### ANSIBLE VERSION

```paste below
2.8
```

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by abadger** [comment]
+1 to feature but we should find a different name.  "Shellisms" and "expand_shell" makes  it sound like we handle all shell syntax but it's really just env vars and tilde.

**Comment by sivel** [comment]
> we should find a different name

I'm game.  I was just using terminology that had been used elsewhere in that function already.

**Comment by bcoca** [comment]
the equivalent expandvars/expanduser functions exist in python already, we can use those names

**Comment by sivel** [comment]
`expand_user_and_vars`?  Kinda wordy.  I'd prefer to not have 2 new args for this.

**Comment by jstoja** [comment]
I think the real terms are "globbling" and "word splitting". Although, I have to admit that (even if personnal opinion) the `expand_shell` name was more self-explanatory than the others.

**Comment by abadger** [comment]
I don't believe it will do globbing in either case.  It may do word splitting but i doubt it (just want to check before asserting that as fact.)

And that's why expand_shell is not a good name ;-)

Fwiw, i agree with everything sivel says https://github.com/ansible/ansible/pull/45620#issuecomment-421147740. I wish there was some better way to list it but if we can't find one, i can live with expand_user_and_vars.

**Comment by sivel** [comment]
I've updated the PR to name the arg `expand_user_and_vars`, as it is the most clear, although a bit too wordy for preference.

**Review by bcoca — APPROVED** [review]

**Comment by abadger** [comment]
+1 from me
